### PR TITLE
Show Facet widgets on taxonomy archives

### LINKED ISSUE #2808: BUG: Facets on taxonomy archives
**Describe the bug**

Currently, it seems the Facets feature is not working on any taxonomy archives, except WooCommerce product categories.

**Steps to Reproduce**
1. Install and enable WooCommerce
2. Activate the WooCommerce and the Facets Feature
3. Sync everything
4. Go to a product category archive
5. See the facets
6. Go to a regular taxonomy archive
7. Do not see facets

**Expected behavior**
Facets would be helpful in any taxonomy, *especially* WooCommerce attributes.

**Additional context**
This might be as simple as adding a `$query->is_tax()` to [this if](https://github.com/10up/ElasticPress/blob/4.2.0/includes/classes/Feature/Facets/Facets.php#L240).

### PR DESCRIPTION
**Pull request by burhandodhy**
### Description of the Change

This PR adds the ability to show the Facet widgets on the taxonomy archives. 

Closes #2808

### Verification Process

- Add the Facet Widget.
- Go to a product category archive
- See the facets

### Changelog Entry

Fixed: Show Facet widgets on taxonomy archives

### Credits

Props @burhandodhy

### CODE REVIEW

**Comment by felipeelia** [comment]
@burhandodhy PHP linting found a problem in the code. Do you mind addressing it and then assigning the PR back to be for review? I think we might be being too simplistic in that new list of checks but I can check that later.

**Comment by burhandodhy** [comment]
@felipeelia liniting issues are fixed. For the check, I replicate the same behaviour which we have before.

**Comment by felipeelia** [comment]
> For the check, I replicate the same behaviour which we have before.

Are you sure that 
```
		if ( ! ( ( function_exists( 'is_product_category' ) && is_product_category() )
			|| $query->is_post_type_archive()
			|| $query->is_search()
			|| ( is_home() && empty( $query->get( 'page_id' ) ) ) )
		) {
```
behaves the same way 
```
if ( $query->is_admin() || $query->is_singular() || $query->is_date() ) {
```
does?

Going to an author archive, just as an example, the code seems to have different behaviors.

Also, it is not needed to check if `$query->is_admin()` because we have a `if ( is_admin() ) {` a couple of lines above that already.

**Comment by burhandodhy** [comment]
@felipeelia thanks for pointing out the author archive page. I totally missed out that case. I updated the check.

File: includes/classes/Feature/Facets/Facets.php [hunk 1]
```diff
@@ -237,15 +237,16 @@ public function is_facetable( $query ) {
...
 		}
 
-		if ( ! ( ( function_exists( 'is_product_category' ) && is_product_category() )
-			|| $query->is_post_type_archive()
+		if ( $query->is_home()
 			|| $query->is_search()
-			|| ( is_home() && empty( $query->get( 'page_id' ) ) ) )
-		) {
-			return false;
+			|| $query->is_tax()
```
Comment (felipeelia) [inline includes/classes/Feature/Facets/Facets.php]: @burhandodhy let's try to pass `get_facetable_taxonomies()` here (and possibly also get rid of is_tag() and is_category())

Comment (burhandodhy) [reply includes/classes/Feature/Facets/Facets.php]: @felipeelia I tried, but `is_tax` doesn't work for the built-in taxonomy page. We don't have any other option other than manually check every page.

`Whether the query is for an existing custom taxonomy archive page. True for custom taxonomy archive pages, false for built-in taxonomies (category and tag archives).`  [source](https://developer.wordpress.org/reference/functions/is_tax/#return)

**Pull request merged**